### DISCUSSION TITLE
Excluded blocks?

### DISCUSSION THREAD

**Discussion by Blaiziuss** [MachiganMC/ProtectedDebugStick#17] [Ideas]
Some blocks can be manipulated to gain an advantage and exploit dupe mechanics, is there a way to edit a list of excluded blocks?

An example from an old 1.8.3 plugin would be:

"COMPOSTER", "CACTUS", "KELP", "SUGAR_CANE", "GRASS_BLOCK", "ACACIA_LEAVES", "AZALEA_LEAVES","BIRCH_LEAVES","DARK_OAK_LEAVES","FLOWERING_AZALEA_LEAVES","JUNGLE_LEAVES","MANGROVE_LEAVES","OAK_LEAVES","SPRUCE_LEAVES", "BIG_DRIPLEAF", "BIG_DRIPLEAF_STEM","CHORUS_FLOWER", "CHORUS_FRUIT", "CHORUS_PLANT", "SEA_PICKLE","RESPAWN_ANCHOR", "ACACIA_TRAPDOOR", "AZALEA_TRAPDOOR","BIRCH_TRAPDOOR","DARK_OAK_TRAPDOOR","FLOWERING_AZALEA_TRAPDOOR","JUNGLE_TRAPDOOR","MANGROVE_TRAPDOOR","OAK_TRAPDOOR","SPRUCE_TRAPDOOR", "WARPED_TRAPDOOR", "CRIMSON_TRAPDOOR","TRIPWIRE_HOOK", "TURTLE_EGG", "BEE_NEST", "BEEHIVE", "CAKE","SWEET_BERRY_BUSH","BEETROOT_SEEDS", "MELON_SEEDS", "PUMPKIN_SEEDS", "WHEAT_SEEDS", "CHEST", "END_PORTAL_FRAME", "END_PORTAL", "PISTON", "PISTON_HEAD", "SNOW"

**Comment by MachiganMC** [comment]
Hello, thank you for the list 😄 
Currently, it exist a blacklist of exluded material you can found [here](https://github.com/MachiganMC/ProtectedDebugStick/blob/904fa861f17bcfc9607f632715813b3322f4fd08/plugin/src/main/resources/config.yml#L197C3-L200C16). 
I can set this as a default blacklist if you think that these blocks it's not adapter for suvival

**Reply by Blaiziuss** [reply]
I'm unsure if setting a default excluded list would be confusing to users who expect a vanilla parity from the tool upon first install. I'm also not sure the exact blocklist that should be blocked (pun intended), I'm not too savvy surrounding what can be exploited with the DS asides from making silly building techniques.

I think whether there is or isn't a block blacklist preconfigured, the option to edit it in the config should be advertised as a feature in the readme; this can apply for the default exception list as well, "ProtectedDebugStick has a block blacklist for exploit protection, the full list can be found and modified at [path/link]".

**Reply by MachiganMC** [reply]
Hey, yes it's better to put an explaination about a "*recommanded blacklist*".
I'll soon make a documentation/wiki and I'll add this in it.

**Comment by Blaiziuss** [comment]
I'm encountering a problem with the blacklist, single blocks listed in the config work fine and aren't able to be edited, however multiple blocks don't work at all. I've tried these various configurations:

-=Works=-
  - CHEST

-=Doesn't Work=-
#on the same line
  - CHEST,COMPOSTER
  - CHEST, COMPOSTER
  - "CHEST","COMPOSTER"
  - "CHEST", "COMPOSTER"
#on individual lines
  - CHEST
  - COMPOSTER

Any ideas? Should the materials be setup as an array?

**Reply by MachiganMC** [reply]
Hello, I test you're blacklist and it works with me.
Does you're config file look like this ?
```yaml
Settings:
  BlackList:
    Material:
      - "CHEST"
      - "COMPOSTER"
```

**Reply by Blaiziuss** [reply]
That works perfectly! I'd just forgotten to add the quotations to the new indents.
